apps/nowcasting-app/components/charts/gsp-pv-remix-chart/index.tsx

```diff
@@ -61,18 +62,65 @@ const GspPvRemixChart: FC<{
     ({} as any);
   const pvPercentage = (forecastAtSelectedTime.expectedPowerGenerationNormalized || 0) * 100;
 
+  const fourHourForecastAtSelectedTime: NonNullable<typeof gsp4HourData>[number] =
```

Comment:
Is this NonNullable thing trying to ensure this is always set? Possibly similar to a bang (!) on the Type..? If so, may need to adjust how this flow works at first glance, as there may be a way for it to error and/or not be set (the `typeof gsp4HourData` can be `undefined`, so not null but not helpful! 

I suspect there was something simple you were trying to sort here that's become something more complex than needed, or maybe I'm misunderstanding!